src/plonk/eval.rs

```diff
@@ -151,6 +170,18 @@ pub trait Eval<F: PrimeField> {
     }
 }
 
+impl<F: PrimeField, E: GetDataForEval<F>> Eval<F, GroupedPoly<F>> for E {
```

Comment:
Do not need to impl Eval for GroupedPoly, instead impl for Expression which is more general and actually needed for verification in is_sat

Reply:
Why switch to `&[Expression]` is bad idea?

To evaluate GroupedPoly, we can call something as simple as this:
```
grouped_poly.iter().map(|cross_term| data.eval(cross_term, row))
```

instead define a wrapper:
```
let res = data.eval(grouped_poly, row) // this is essentially above code
```

Reply:
Reducing the context does not do any good in this case

For example, knowing for example the fact that for some degree the coefficient is zero will allow you not to count it. 
https://github.com/snarkify/sirius/blob/da64ae419bfd63c8859d0269790562906048796b/src/plonk/eval.rs#L206

Reply:
Especially `GroupedPoly` already implements `fn iter`, so there are no disadvantages to the approach with additional wrapper, your approach can be implemented by it, however, with the presence of additional context different optimizations are possible